File: library/src/test/java/io/github/anderscheow/validator/rules/common/NotNullRuleTest.java
package io.github.anderscheow.validator.rules.common;

import android.support.annotation.StringRes;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.Locale;

import static org.junit.Assert.*;

public class NotNullRuleTest {

    private NotNullRule notNullRule;

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
        notNullRule = null;
    }

    @Test
    public void validate_ValidSample_ReturnTrue() throws Exception {
        notNullRule = new NotNullRule();

        String sample = "test";

        assertTrue(notNullRule.validate(sample));
    }

    @Test
    public void validate_InvalidSample_ReturnFalse() throws Exception {
        notNullRule = new NotNullRule();

        assertFalse(notNullRule.validate(null));
    }

    @Test
    public void errorMessage_DefaultErrorMessage() throws Exception {
        String errorMessage = "Value must not be null";

        notNullRule = new NotNullRule();

        assertEquals(errorMessage, notNullRule.errorMessage());
    }

    @Test
    public void errorMessage_CustomErrorMessage() throws Exception {
        String errorMessage = "This is a custom error message";

        notNullRule = new NotNullRule(errorMessage);

        assertEquals(errorMessage, notNullRule.errorMessage());
    }

    @Test
    public void errorMessage_CustomErrorRes() throws Exception {
        @StringRes int errorRes = 0;

        notNullRule = new NotNullRule(errorRes);

        assertEquals(errorRes, notNullRule.errorRes());
    }
}

File: library/src/test/java/io/github/anderscheow/validator/rules/common/NotNullRuleTest.kt
package io.github.anderscheow.validator.rules.common

import androidx.annotation.StringRes
import org.junit.After
import org.junit.Assert.*
import org.junit.Before
import org.junit.Test

class NotNullRuleTest {

    private lateinit var notNullRule: NotNullRule

    @Before
    @Throws(Exception::class)
    fun setUp() {
    }

    @After
    @Throws(Exception::class)
    fun tearDown() {
    }

    @Test
    @Throws(Exception::class)
    fun validate_ValidSample_ReturnTrue() {
        notNullRule = NotNullRule()

        val sample = "test"

        assertTrue(notNullRule.validate(sample))
    }

    @Test
    @Throws(Exception::class)
    fun validate_InvalidSample_ReturnFalse() {
        notNullRule = NotNullRule()

        assertFalse(notNullRule.validate(null))
    }

    @Test
    @Throws(Exception::class)
    fun errorMessage_DefaultErrorMessage() {
        val errorMessage = "Value must not be null"

        notNullRule = NotNullRule()

        assertEquals(errorMessage, notNullRule.errorString)
    }

    @Test
    @Throws(Exception::class)
    fun errorMessage_CustomErrorMessage() {
        val errorMessage = "This is a custom error message"

        notNullRule = NotNullRule(errorMessage)

        assertEquals(errorMessage, notNullRule.errorString)
    }

    @Test
    @Throws(Exception::class)
    fun errorMessage_CustomErrorRes() {
        @StringRes val errorRes = 0

        notNullRule = NotNullRule(errorRes)

        assertEquals(errorRes, notNullRule.errorRes)
    }
}